From add99ede2499d69501d04aa379512b01c00bd098 Mon Sep 17 00:00:00 2001
From: Hardik Shah <hardik.shah@ti.com>
Date: Mon, 9 Mar 2009 16:39:42 +0530
Subject: [PATCH 333/426] Crash fixed in FBDEV power managent.

FBDEV was calling disable panel without
checking for the display and manager structure.
---
 drivers/video/omap2/omapfb-sysfs.c |   15 +++++++++++----
 1 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/drivers/video/omap2/omapfb-sysfs.c b/drivers/video/omap2/omapfb-sysfs.c
index 4dfac97..6214130 100755
--- a/drivers/video/omap2/omapfb-sysfs.c
+++ b/drivers/video/omap2/omapfb-sysfs.c
@@ -951,15 +951,22 @@ void omap2fb_workqueue_handler(struct work_struct *work)
 
 	if (omap2fb_can_sleep == 1) {
 		for (i = 0; i < omap2fb->num_fbs; i++) {
-			display = omap2fb->overlays[i]->manager->display;
-			display->disable(display);
+			if (omap2fb->overlays[i]->manager &&
+					omap2fb->overlays[i]->manager->display) {
+				display =
+					omap2fb->overlays[i]->manager->display;
+				display->disable(display);
+			}
 		}
 		omap2fb_can_sleep = 2;
 		del_timer(&omap2fb->timer);
 	} else if (omap2fb_can_sleep == 3){
 		for (i = 0; i < omap2fb->num_fbs; i++) {
-			display = omap2fb->overlays[i]->manager->display;
-			display->enable(display);
+			if (omap2fb->overlays[i]->manager &&
+					omap2fb->overlays[i]->manager->display) {
+				display = omap2fb->overlays[i]->manager->display;
+				display->enable(display);
+			}
 		}
 		omap2fb_can_sleep = 0;
 		mod_timer(&omap2fb->timer, jiffies + omap2fb->sleep_timeout);
-- 
1.6.2.4

